### PR TITLE
Add PKey docs

### PR DESCRIPTION
**Pull request by apeduru**
Refs [#716](https://github.com/sfackler/rust-openssl/issues/716)

### CODE REVIEW

File: openssl/src/pkey.rs [hunk 1]
```diff
@@ -1,3 +1,44 @@
...
+//! * Elliptic Curves
+//!
+//! These algorithms rely on hard mathematical problems - namely integer factorization,
+//! discrete logarithms, and elliptic curve relationships - that currently do not
+//! yield efficient solutions. This property ensures the security of these
+//! cryptographic algorithms.
+//!
+//! # Example
+//!
+//! Generate a 2048-bit RSA public/private key pair.
```
Comment (sfackler) [inline openssl/src/pkey.rs]: Not sure this example is really showing much.

Comment (apeduru) [reply openssl/src/pkey.rs]: I can add another couple statements serializing the pub/priv key pair to PEM. Did you have something else in mind? I can also spin it out into a separate example.

Comment (sfackler) [reply openssl/src/pkey.rs]: I don't really have any particular example in mind for this module. `PKey` doesn't really do much on its own.

**Comment by apeduru** [comment]
not sure why the macos build failed

File: openssl/src/pkey.rs [hunk 2]
```diff
@@ -1,3 +1,48 @@
+//! Public/private key processing.
+//!
+//! Asymmetric public key algorithms solve the problem of establishing and sharing
+//! secret keys to securely send and receive messages.
+//! This system uses a pair of keys: a public key, which can be freely
+//! distributed, and a private key, which is kept to oneself. An entity may
+//! encrypt information using a user's public key. The encrypted information can
+//! only be deciphered using that user's private key.
+//!
+//! This module offers support for four popular algorithms:
```
Comment (sfackler) [inline openssl/src/pkey.rs]: HMAC is also mixed in here.

**Comment by sfackler** [comment]
Yeah, don't worry about the OSX failure.

Looks good to me other than the one note about HMAC!

**Pull request merged**